components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,467 @@
+	end
+end
+
+--- Fetches the LPDB object of a tournament
+function PrizePool._getTournamentInfo(pageName)
+	return mw.ext.LiquipediaDB.lpdb('tournament', {
+		conditions = '[[pagename::' .. pageName .. ']]',
+		limit = 1,
+	})[1]
+end
+
+--- Get the default date based on wiki-variables set in the Infobox League
+function PrizePool._getTournamentDate()
+	return Variables.varDefaultMulti('tournament_enddate', 'tournament_edate', 'edate', TODAY)
+end
+
+---
+function Placement:init(args, placeStart, opponentType)
+	self.args = self:_parseArgs(args)
+	self.date = self.args.date or PrizePool._getTournamentDate()
+	self.placeStart = self.args.placeStart
+	self.placeEnd = self.args.placeEnd
+	self.opponentType = opponentType
+	self.opponents = {}
+	self.prizes = {}
+
+	-- Parse prizes
+	self.prizes = self:_readPrizeRewards(self.args)
+
+	-- Parse opponents
+	self.opponents = self:_parseOpponents(self.args)
+
+	-- Implicit place range has been given (|place= is not set)
+	-- Use the last known place and set the place range based on the entered number of opponents
+	if not self.placeStart and not self.placeEnd then
+		self.placeStart = placeStart
+		self.placeEnd = placeStart + #self.opponents - 1
+	end
+
+	assert(self.placeStart and self.placeEnd, 'readPlacements: Invalid |place= provided.')
+	assert(#self.opponents > self.placeEnd - self.placeStart, 'readPlacements: Too many opponents')
+end
+
+function Placement:_parseArgs(args)
+	local parsedArgs = Table.deepCopy(args)
+	-- Explicit place range has been given
+	if args.place then
+		local places = Table.mapValues(mw.text.split(args.place, '-'), tonumber)
+		parsedArgs.placeStart = places[1]
+		parsedArgs.placeEnd = places[2] or places[1]
+	end
+	return parsedArgs
+end
+
+--- Parse the input for available rewards of prizes, for instance how much money a team would win.
+function Placement:_readPrizeRewards(args)
+	local rewards = {}
+
+	for prizeEnum in pairs(prizeTypes) do
+		local prizeDatum = prizeData[prizeTypes[prizeEnum]]
+		local fieldName = prizeDatum.row
+		if fieldName then
+			args[fieldName .. '1'] = args[fieldName .. '1'] or args[fieldName]
+			for _, rewardValue, index in Table.iter.pairsByPrefix(args, fieldName) do
+				if prizeTypes[prizeEnum] == prizeTypes.USD then
+					self.hasUSD = true
+				end
+				rewards[prizeEnum .. index] = prizeDatum.rowParse(self, rewardValue, args, index)
+			end
+		end
+	end
+
+	return rewards
+end
+
+function Placement:_readAdditionalData(args)
+	local data = {}
+
+	for enum in pairs(additionalDataTypes) do
+		local type = additional[additionalDataTypes[enum]]
+		local fieldName = type.field
+		if args[fieldName] then
+			data[enum] = type.parse(self, args[fieldName], args)
+		end
+	end
+
+	return data
+end
+
+function Placement:_parseOpponents(args)
+	-- Parse opponents in the placement
+	for opponentIndex = 1, math.huge do
+		local opponentInput = Json.parseIfString(args[opponentIndex])
+		local opponent = {opponentData = {}, prizes = {}, data = {}}
```

Comment:
```
opponents[i]
opponents[i].opponentData
opponents[i].data
```

the names for those 3 objects don't succinctly represent what's inside the objects

I ran into the same problem when creating PrizePool/next - the solution there was:
```
entries[i]
entries[i].opponent
```
and merge `wdl, vsOpponent, etc` fields in `entries[i]` itself.

Reply:
Not opposed to this

Reply:
Keeping this out for now. With `data` renamed `additonalData` and prizes `prizeRewards` I think it's clearer what each of the tables contain.